### PR TITLE
STYLE: replace unwanted-patterns-strings-to-concatenate with ruff

### LINKED ISSUE #51406: STYLE: replace unwanted-patterns-strings-to-concatenate with ruff
Task here is:

1. in 

  https://github.com/pandas-dev/pandas/blob/fe85cbfb20f3f360ebba62905b93a6f98bf655c1/pyproject.toml#L201-L215

  add
```
# implicit string concatenation
"ISC",
```

2. run `pre-commit run ruff --all-files`, and fix up everything it flags
3. remove 

  https://github.com/pandas-dev/pandas/blob/fe85cbfb20f3f360ebba62905b93a6f98bf655c1/.pre-commit-config.yaml#L348-L352

4. remove

https://github.com/pandas-dev/pandas/blob/73840ef221050af2588b079bd6ba45c8e291bcba/scripts/validate_unwanted_patterns.py#L235-L281

5. remove 

https://github.com/pandas-dev/pandas/blob/3a0db109390ddf9ab331cc86e63449c7445aa8a7/scripts/tests/test_validate_unwanted_patterns.py#L156-L195

6. if there's anything else related to this old script which `ruff` now supplants, feel free to remove that too

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #51406

`Unwanted-patterns-strings-to-concatenate` replaced with ruff.

### CODE REVIEW

File: asv_bench/benchmarks/io/csv.py [hunk 1]
```diff
@@ -318,7 +318,8 @@ def setup(self, sep, decimal, float_precision):
             "".join([random.choice(string.digits) for _ in range(28)])
             for _ in range(15)
         ]
-        rows = sep.join([f"0{decimal}" + "{}"] * 3) + "\n"
+        row = "".join([f"0{decimal}", "{}"])
```
Comment (MarcoGorelli) [inline asv_bench/benchmarks/io/csv.py:321]: Thanks for working on this! 

Would it work to do
```python
rows = sep.join([f"0{decimal}{{}}"] * 3) + "\n"
```
?

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -2181,18 +2181,18 @@ def test_to_string_float_na_spacing(self):
...
+            "4       NaN"
         )
         assert result == expected
 
     def test_to_string_without_index(self):
         # GH 11729 Test index=False option
         s = Series([1, 2, 3, 4])
         result = s.to_string(index=False)
-        expected = "1\n" + "2\n" + "3\n" + "4"
+        expected = "".join(["1\n", "2\n", "3\n", "4"])
```
Comment (twoertwein) [inline pandas/tests/io/formats/test_format.py:2195]: Could probably be simplified as either `"1\n2\n3\n4"` or `"\n".join(["1", "2", "3"])`

(similar for the next change)

File: pandas/tests/io/parser/conftest.py [hunk 3]
```diff
@@ -189,7 +189,9 @@ def all_parsers_all_precisions(request):
 _encoding_prefixes = ["utf", "UTF"]
 
 _encoding_fmts = [
-    f"{prefix}{sep}" + "{0}" for sep in _encoding_seps for prefix in _encoding_prefixes
+    "{}{}".format(f"{prefix}{sep}", "{0}")
```
Comment (twoertwein) [inline pandas/tests/io/parser/conftest.py:192]: could probably be `f"{prefix}{sep}{{0}}"`

File: asv_bench/benchmarks/io/csv.py (hunk 1 above)
Comment (natmokval) [reply asv_bench/benchmarks/io/csv.py]: Thank you, it works nicely. I corrected my PR.

File: pandas/tests/io/formats/test_format.py (hunk 2 above)
Comment (natmokval) [reply pandas/tests/io/formats/test_format.py]: Thank you, I did as you suggested. It works very well.

**Review by twoertwein — APPROVED** [review]
There are some `"".join(...)` calls which could be simplified but I also wouldn't mind keeping them as-is (that way it might be clearer that they represent different rows in the Series/DataFrame).

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me on green, do you want to mark as ready for review?

**Comment by natmokval** [comment]
I resolved the merge conflict and updated my branch.

**Comment by twoertwein** [comment]
Thank you @natmokval!

**Pull request merged**